tiatoolbox/models/architecture/unet.py

```diff
@@ -220,20 +226,33 @@ def __init__(
         num_input_channels: int = 2,
         num_output_channels: int = 2,
         encoder: str = "resnet50",
-        decoder_block: Tuple[int] = (3, 3),
+        encoder_levels: List[int] = None,
+        decoder_block: Tuple[int] = None,
+        skip_type: str = "add",
     ):
         super().__init__()
 
+        if encoder.lower() not in {"resnet50", "unet"}:
+            raise ValueError(f"Unknown encoder `{encoder}`")
+
+        if encoder_levels is None:
+            encoder_levels = [64, 128, 256, 512, 1024]
+
+        if decoder_block is None:
+            decoder_block = [3, 3]
+
         if encoder == "resnet50":
-            padding = 1
+            padding = "same"
             preact = True
             self.backbone = ResNetEncoder.resnet50(num_input_channels)
         elif encoder == "unet":
-            padding = 0
+            padding = "same"
             preact = False
```

Comment:
Back to above about hard locking decision. Should fix the padding below within the decoder creation rather than using "same". Additionally, do not use "same" or "valid" to maintain backward compatibility (only supported 1.9.0 onward)
https://pytorch.org/docs/1.8.0/_modules/torch/nn/modules/conv.html#Conv2d

Reply:
Fixing within `create_block`

Reply:
done in 7da8262